src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/IntervalTree`1.cs

```diff
@@ -50,6 +48,11 @@ public static IntervalTree<T> Create<TIntrospector>(in TIntrospector introspecto
         return result;
     }
 
+    /// <summary>
+    /// Provides access to lots of common algorithms on this interval tree.
+    /// </summary>
+    public IntervalTreeAlgorithms<T, BinaryIntervalTree<T>> Algorithms => new(this);
```

Comment:
this is how we get type inference to work.  a user just calls .Algorithms, and the T and TIntervalTree types are already solved for them on the Algorithms wrapper they get back.